qml: Add DropShadow to the OptionSwitch indicatorButton

shaavan opened the pull request:
- This PR is a follow-up to #101 and addresses the suggestions given there after it was merged.
- This PR also adds the **DropShadow** for the **OptionSwitch**'s `indicatorButton`, as per the [main design file](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App-Main?node-id=1035%3A1883) of the project.

Some points to note:

- Used `===` instead of `==` to avoid type coercion.
- Chose sample value based on official recommended guidelines (sample = 2*radius + 1) https://doc.qt.io/qt-5/qml-qtgraphicaleffects-dropshadow.html#samples-prop
- Used rgba to hex convertor for shadow color: https://rgbacolorpicker.com/rgba-to-hex

shaavan commented:
Running the current version of the PR will cause the following error.

```
qrc:/qml/controls/OptionSwitch.qml:7:1: module "QtGraphicalEffects" is not installed
```

And to create _dropShadow_, we require this module. I tried to figure out some workaround, but I thought it's better to ask and go with the correct approach on the get-go.

shaavan commented:
Updated from e2d1635 to 2e8b096.

Changes:
- Made appropriate styling changes.


**Diff:**

```diff
diff --git a/src/qml/components/ConnectionSettings.qml b/src/qml/components/ConnectionSettings.qml
index 090e8e793..35f8e5d3a 100644
--- a/src/qml/components/ConnectionSettings.qml
+++ b/src/qml/components/ConnectionSettings.qml
@@ -15,21 +15,21 @@ ColumnLayout {
     }
     Setting {
         Layout.fillWidth: true
-        header:  qsTr("Daily upload limit")
+        header: qsTr("Daily upload limit")
     }
     Setting {
         Layout.fillWidth: true
-        header:  qsTr("Connection limit")
+        header: qsTr("Connection limit")
     }
     Setting {
         Layout.fillWidth: true
-        header:  qsTr("Listening enabled")
-        description:  qsTr("Reduces data usage.")
+        header: qsTr("Listening enabled")
+        description: qsTr("Reduces data usage.")
     }
     Setting {
         last: true
         Layout.fillWidth: true
-        header:  qsTr("Blocks Only")
-        description:  qsTr("Do not transfer unconfirmed transactions. Also disabled listening.")
+        header: qsTr("Blocks Only")
+        description: qsTr("Do not transfer unconfirmed transactions. Also disabled listening.")
     }
 }

```

jarolrod commented on src/qml/controls/OptionSwitch.qml:
```diff
@@ -4,14 +4,15 @@
         implicitHeight: 28
         x: root.leftPadding
-        y: parent.height / 2 - height / 2
+        y: Math.round((parent.height - height) / 2)
```
Follow ups to #101 should be in their own PR with well structured and focused commits.

shaavan replied:
Addressed in 109682b971e08bf282b82b7362c136f68340abee

jarolrod commented on src/qml/controls/OptionSwitch.qml:
```diff
@@ -23,5 +24,16 @@ Switch {
             radius: 18
             color: "#ffffff"
         }
+
```
```suggestion
```

shaavan replied:
Addressed in #112

promag commented:
> Running the current version of the PR will cause the following error.
> 
> ```
> qrc:/qml/controls/OptionSwitch.qml:7:1: module "QtGraphicalEffects" is not installed
> ```

@shaavan try adding

```cpp
Q_IMPORT_PLUGIN(QtGraphicalEffectsPlugin)
Q_IMPORT_PLUGIN(QtGraphicalEffectsPrivatePlugin)
```

https://github.com/bitcoin-core/gui-qml/blob/ef98ffa9e0a1054d7c910221b76bd13d1a81ef28/src/qml/bitcoin.cpp#L43-L52

shaavan commented:
Updated from 2e8b096 to 109682b ([pr111.01](https://github.com/shaavan/gui-qml/tree/pr111.01) -> [pr112.02](https://github.com/shaavan/gui-qml/tree/pr111.02))
Addressed @jarolrod, @promag suggestions.

Changes:
- Follow-ups related to #101 have been removed from this PR. The stated follow-ups were merged in #112.
- Used **Q_IMPORT_PLUGIN** to import relevant modules in `src/qml/bitcoin.cpp` file.


The PR is still in WIP. Currently, using the **Q_IMPORT_PLUGIN** caused the following error.

```
/usr/bin/ld: qt/libbitcoinqt.a(qt_libbitcoinqt_a-bitcoin.o): in function `_GLOBAL__sub_I__Z10QmlGuiMainiPPc':
bitcoin.cpp:(.text.startup+0xc2): undefined reference to `qt_static_plugin_QtGraphicalEffectsPlugin()'
/usr/bin/ld: bitcoin.cpp:(.text.startup+0xd2): undefined reference to `qt_static_plugin_QtGraphicalEffectsPrivatePlugin()'
collect2: error: ld returned 1 exit status
```

jarolrod reviewed (commented):
@shaavan as properly supporting this requires OpenGL support, let's leave the dropshadows for later.

shaavan commented:
> supporting this requires OpenGL support, let's leave the dropshadows for later.

I understand. I shall close this PR for now. We can restart work on this once OpenGL support is appropriately integrated.
